qiskit/algorithms/evolvers/pvqd.py

```diff
@@ -0,0 +1,472 @@
+                previous iteration result is used as initial guess. If None, this is set to
+                a random vector with elements in the interval :math:`[-0.01, 0.01]`.
+            evolution: The evolution synthesis to use for the construction of the Trotter step.
+                Defaults to first-order Lie-Trotter decomposition.
+            gradients: If True, use the parameter shift rule to compute gradients. If False,
+                the optimizer will not be passed a gradient callable.
+            quantum_instance: The backend of quantum instance used to evaluate the circuits.
+        """
+        if evolution is None:
+            evolution = LieTrotter()
+
+        self.ansatz = ansatz
+        self.initial_parameters = initial_parameters
+        self.timestep = timestep
+        self.optimizer = optimizer
+        self.initial_guess = initial_guess
+        self.expectation = expectation
+        self.evolution = evolution
+        self.gradients = gradients
+
+        self._sampler = None
+        self.quantum_instance = quantum_instance
+
+    @property
+    def quantum_instance(self) -> Optional[QuantumInstance]:
+        """Return the current quantum instance."""
+        return self._quantum_instance
+
+    @quantum_instance.setter
+    def quantum_instance(self, quantum_instance: Optional[Union[Backend, QuantumInstance]]) -> None:
+        """Set the quantum instance and circuit sampler."""
+        if quantum_instance is not None:
+            if not isinstance(quantum_instance, QuantumInstance):
+                quantum_instance = QuantumInstance(quantum_instance)
+            self._sampler = CircuitSampler(quantum_instance)
+
+        self._quantum_instance = quantum_instance
+
+    def step(
+        self, hamiltonian: OperatorBase, theta: np.ndarray, dt: float, initial_guess: np.ndarray
+    ) -> Tuple[np.ndarray, float]:
+        """Perform a single time step.
+
+        Args:
+            hamiltonian: The Hamiltonian under which to evolve.
+            theta: The current parameters.
+            dt: The time step.
+            initial_guess: The initial guess for the classical optimization of the
+                fidelity between the next variational state and the Trotter-evolved last state.
+                If None, this is set to a random vector with elements in the interval
+                :math:`[-0.01, 0.01]`.
+
+        Returns:
+            A tuple consisting of the next parameters and the fidelity of the optimization.
+        """
+        # construct cost function
+        loss, gradient = self.get_loss(hamiltonian, dt, theta)
+
+        if initial_guess is None:
+            initial_guess = np.random.random(self.initial_parameters.size) * 0.01
+
+        # call optimizer
+        if isinstance(self.optimizer, Optimizer):
+            optimizer_result = self.optimizer.minimize(loss, initial_guess, gradient)
+        else:
+            optimizer_result = self.optimizer(loss, initial_guess, gradient)
```

Comment:
```suggestion
        if isinstance(self.optimizer, Optimizer):
            optimizer_result = self.optimizer.minimize(loss, initial_guess, gradient)
        else:
            optimizer_result = self.optimizer(loss, initial_guess, gradient)
```
It is rather self-explanatory.